sdk/storage/azure-storage-queue/azure/storage/queue/_encryption.py

```diff
@@ -837,10 +876,12 @@ def decrypt_blob(  # pylint: disable=too-many-locals,too-many-statements
         total_size = len(content)
         offset = 0
 
-        nonce_length = encryption_data.encrypted_region_info.nonce_length
-        data_length = encryption_data.encrypted_region_info.data_length
-        tag_length = encryption_data.encrypted_region_info.tag_length
-        region_length = nonce_length + data_length + tag_length
+        if encryption_data.encrypted_region_info is not None:
+            if isinstance(encryption_data.encrypted_region_info, _EncryptedRegionInfo):
```

Comment:
`isinstance` check is unnecessary here, but will have `None` check as `encryption_region_info` is only used in V2.

In general, anytime we use `encryption_region_info` or `content_encryption_IV` we are going to have to null check as they are Optional (since one is for encryption V2 and the other is V1 respectively)

In this case, do you want to throw on a failed `None` check? If so, I can go back and apply this to these cases.

Reply:
Let's go with if is None:
-> throw ValueError "Where is your encrypted region info" 😄